packages/tpetra/core/src/Tpetra_DirectoryImpl_def.hpp

```diff
@@ -431,10 +431,28 @@ namespace Tpetra {
 
       RCP<const Teuchos::Comm<int> > comm = map.getComm ();
       const int numProcs = comm->getSize ();
-      const global_size_t nOverP = map.getGlobalNumElements () / numProcs;
       const LO LINVALID = Teuchos::OrdinalTraits<LO>::invalid();
       LookupStatus res = AllIDsPresent;
 
+      // Find the first initialized GID for search below
+      auto it = std::find_if(
```

Comment:
This should be OK, since line 303 of `Tpetra_DirectoryImpl_def.hpp` always initializes `allMinGIDs_` to size `numProcs + 1`.